### DISCUSSION TITLE
Linking a static library with kbuild

### DISCUSSION THREAD

**Discussion by whoenig** [bitcraze/discussions#1236] [Q&A]
I am trying to update the rust app example from https://github.com/ataffanel/crazyflie-app-hello-rs to the latest firmware. The major change is the switch from Makefile to Kbuild. However, I can't get Kbuild to actually link against the static rust library.

Specifically, the tips in https://stackoverflow.com/questions/56926632/how-do-i-link-a-static-library-when-building-a-kernel-module don't work. There also seems to be some preparation in https://github.com/bitcraze/crazyflie-firmware/blob/f61da11d54b6d54c7fa746688e8e9be4edd73a29/tools/kbuild/Makefile.kbuild#L559 for libs, but it's never used and doesn't work for me either.

All attempts are in https://github.com/IMRCLab/crazyflie-app-hello-rs/blob/main/Kbuild. If I manually hack the Makefile.kbuild file to include the *.a it works, so this question is really about how to "properly" do it, ideally without changing the firmware itself.

Related to https://github.com/orgs/bitcraze/discussions/1192.

**Comment by ataffanel** [comment]
I am not sure how to do it properly but this is something we should definitely look at and figure out. I have created a ticker for it since this is something that will need to be implemented in the firmware repos: https://github.com/bitcraze/crazyflie-firmware/issues/1354. Hopefully I will get some time in the following weeks to look at it, it would be very nice to have a proper way to add some rust in the firmware :-).
